### PR TITLE
Feature/centralized dynamicparam

### PR DESCRIPTION
**Pull request by tosoikea**
1. Addition of Get-LoggingDynamicParam
+ Centralized function for creating dictionaries used inside DynamiParam {}
+ Improved readability for Add-LoggingTarget, Set-LoggingDefaultLevel, Write-Log

2. Small addition to Set-LoggingDefaultLevel and the discarding of lower logging level messsages

### CODE REVIEW

**Comment by renehernandez** [comment]
@tosoikea Is there any issue associated with this pull request?

**Comment by tosoikea** [comment]
@renehernandez No associated issue. Just enforcing modularization to use dynamic param for future methods and better readability of existing functions.

File: Logging/private/Get-LoggingDynamicParam.ps1 [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    Get-LoggingDynamicParam -Name "Level" -Level -DefaultValue 'Verbose'
+}
+
+DynamicParam{
+    $dictionary = Get-LoggingDynamicParam -Name "Level" -Level
+    Get-LoggingDynamicParam -Name "Target" -Target -DynamicParams $dictionary
+}
+#>
+
+function Get-LoggingDynamicParam {
```
Comment (renehernandez) [inline Logging/private/Get-LoggingDynamicParam.ps1]: Using `Get` as verb here does not seem appropriate. Get is usually for retrieving already existing data or objects. 

I believe a better name would be `New-LoggingDynamicParam`

File: Logging/private/Get-LoggingDynamicParam.ps1 [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    [OutputType([System.Management.Automation.RuntimeDefinedParameterDictionary])]
+    [CmdletBinding()]
```
Comment (renehernandez) [inline Logging/private/Get-LoggingDynamicParam.ps1]: You should specify a DefaultParameterSetName in the `CmdletBinding` attribute

**Comment by renehernandez** [comment]
@tosoikea Thanks for the contribution.

I think we would have been better off by first having a discussion through a new issue about how and why this would improve the state of code

**Comment by tosoikea** [comment]
@renehernandez I agree. But I hope you also agree that copy and pasting the same code three times is not a good practice.

File: Logging/private/Get-LoggingDynamicParam.ps1 (hunk 1 above)
Comment (tosoikea) [reply Logging/private/Get-LoggingDynamicParam.ps1]: Thank you. I will apply the recommendations tomorrow. (no laptop till then)

**Comment by tosoikea** [comment]
@renehernandez I changed the files accordingly.

**Comment by EsOsO** [comment]
Wow guys, amazing work ;)
Never thought a little project like this could bring people to contribute and discuss on how to move in the right direction.

I'll ask you only to keep the documentation updated. 

Feel free to contribute and propose new features. 
Keep up the good work.

**Pull request merged**